source/grid/grid_generator.cc

```diff
@@ -7250,29 +7363,101 @@ namespace GridGenerator
         // create cells one by one
 
         if (dim == 2)
-          for (const auto &cell : table_2D_cell)
-            add_cell(dim, cell, current_material_id);
+          {
+            const auto current_manifold_id_of_cell =
+              cell.manifold_id(); // get cell-manifold id from current quad cell
+            for (const auto &cell : table_2D_cell)
+              add_cell(dim,
+                       cell,
+                       current_material_id,
+                       current_manifold_id_of_cell); // set cell-manifold id
```

Comment:
Drop the comment. Since it's after the whole call, it looks like you're commenting on what the function call here does, but I think what you're really doing is only commenting on the last line -- but since the previous lines don't have such comments, this looks out of place.